blockmanager.go

```diff
@@ -259,7 +259,16 @@ func (b *blockManager) Start() {
 	log.Trace("Starting block manager")
 	b.wg.Add(2)
 	go b.blockHandler()
-	go b.cfHandler()
+	go func() {
+		defer b.wg.Done()
+
+		select {
+		case <-firstPeerConnect:
+			b.cfHandler()
```

Comment:
I would also add debug logs `"Waiting for peer connection..."` and `"Peer connected, starting cfHandler..."`

Reply:
good suggestion, added.